### PR TITLE
GNSS Quectel LCX6G improve PM and stability

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
* Improve power management of the LCX6G device driver, allowing it to properly handle being on a power domain, and reacting appropriately to the power domain going on and off.
* Close the modem_pipe after suspending to flush it before resuming
* Created Kconfigs for configurable buffers
* Improved logging

Tested with this while loop
```
	for (uint8_t i = 0; i < 255; i++) {
		if (pm_device_runtime_get(DEVICE_DT_GET(DT_ALIAS(gnss))) < 0) {
			printf("get failed at %u\n", i);
			return 0;
		}
		if (pm_device_runtime_put(DEVICE_DT_GET(DT_ALIAS(gnss))) < 0) {
			printf("put failed at %u\n", i);
			return 0;
		}
	}

	printf("done\n");
	return 0;
```
on an STM32 based chip with the LC76G on a power domain, and with a nordic based chip with the LC76G constantly powered.

### CODE REVIEW

**Review by rerickson1 — DISMISSED** [review]

File: drivers/gnss/gnss_quectel_lcx6g.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ struct quectel_lcx6g_data {
 	/* UART backend */
 	struct modem_pipe *uart_pipe;
 	struct modem_backend_uart uart_backend;
-	uint8_t uart_backend_receive_buf[128];
+	uint8_t uart_backend_receive_buf[126];
```
Comment (ycsin) [inline drivers/gnss/gnss_quectel_lcx6g.c]: this change looks off, is it intended?

**Review by ycsin — COMMENTED** [review]
looks like it should have been 256, based on the Kconfig default value in the next commit?

File: drivers/gnss/gnss_quectel_lcx6g.c (hunk 1 above)
Comment (bjarki-trackunit) [reply drivers/gnss/gnss_quectel_lcx6g.c]: This specific change was not intended, but increasing the default buffer size to 256 with the Kconfig is :)

Comment (bjarki-trackunit) [reply drivers/gnss/gnss_quectel_lcx6g.c]: I updated the commit message to state the intentional increase in buffer size :)

File: drivers/gnss/gnss_quectel_lcx6g.c [hunk 2]
```diff
@@ -22,10 +22,8 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(quectel_lcx6g, CONFIG_GNSS_LOG_LEVEL);
 
-#define QUECTEL_LCX6G_STARTUP_DELAY           (K_MSEC(300U))
-#define QUECTEL_LCX6G_STATE_CHANGE_DELAY_MSEC (300LL)
-#define QUECTEL_LCX6G_PAIR_TIMEOUT            (K_SECONDS(11))
-#define QUECTEL_LCX6G_SCRIPT_TIMEOUT_S        (10U)
+#define QUECTEL_LCX6G_PM_TIMEOUT_MS    (500U)
+#define QUECTEL_LCX6G_SCRIPT_TIMEOUT_S (10U)
```
Comment (fabiobaltieri) [inline drivers/gnss/gnss_quectel_lcx6g.c:26]: hey do you really want to keep these parenthesized? there's no real need for it and they look a bit odd to me

**Review by fabiobaltieri — DISMISSED** [review]

File: drivers/gnss/gnss_quectel_lcx6g.c (hunk 2 above)
Comment (bjarki-trackunit) [reply drivers/gnss/gnss_quectel_lcx6g.c]: what parantheses :D

**Comment by bjarki-trackunit** [comment]
removed redundant parentheses :)

**Review by fabiobaltieri — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]

**Pull request merged**